### PR TITLE
Fix compilation errors on FreeBSD.

### PR DESCRIPTION
**Pull request by josteink**
Without this fix the build breaks at the following error:

    make[1]: stopped in /usr/home/josteink/build/coreclr/binaries/intermediates/freebsd.x64.debug
    /home/josteink/build/coreclr/src/pal/tools/cppmunge/cppmunge.c:39:10: fatal error: 'linux/limits.h' file not found
    #include <linux/limits.h>
             ^
    1 error generated.

### CODE REVIEW

File: src/debug/debug-pal/CMakeLists.txt [hunk 1]
```diff
@@ -25,6 +25,10 @@ if(CLR_CMAKE_PLATFORM_UNIX)
       unix/dynamiclibaddress.cpp
     )
 
+  if(CLR_CMAKE_PLATFORM_FREEBSD)
+    add_definitions(-D__FREEBSD__=1)
```
Comment (emaste) [inline src/debug/debug-pal/CMakeLists.txt]: For FreeBSD the compiler defines `__FreeBSD__` by default

Comment (josteink) [inline src/debug/debug-pal/CMakeLists.txt]: Didn't know that. Pardon my cmake :)

I'll take that into account and tighten up the code accordingly.

File: src/pal/tools/cppmunge/CMakeLists.txt [hunk 2]
```diff
@@ -10,6 +10,10 @@ project(cppmunge)
 
 add_definitions(-D_BSD_SOURCE) # Needed to get realpath part of CRT for c99 compat.
 
+if(CLR_CMAKE_PLATFORM_LINUX)
+  add_definitions(-D__LINUX__=1)
```
Comment (emaste) [inline src/pal/tools/cppmunge/CMakeLists.txt]: similarly I think there is __linux__ or __Linux__ by default on Linux platforms

Comment (josteink) [inline src/pal/tools/cppmunge/CMakeLists.txt]: I'm not too sure about that. Anyway, I just followed the general methodology from the base CMakeLists.txt in the repo:

https://github.com/dotnet/coreclr/blob/master/CMakeLists.txt#L350-L354

Comment (jkotas) [inline src/pal/tools/cppmunge/CMakeLists.txt]: cppmunge is dead code. I have submitted PR to delete it.

Comment (emaste) [inline src/pal/tools/cppmunge/CMakeLists.txt]: For reference, here's an example of use in LLDB:
https://github.com/llvm-mirror/lldb/blob/master/source/Target/Platform.cpp#L572

**Comment by josteink** [comment]
@jkotas It could be. As a relatively new collaborator I just wanted to make sure my patch the smallest possible impact on the code being compiled, so I went with the absolute bare minimum.

It's your call really.

File: src/pal/tools/cppmunge/CMakeLists.txt (hunk 2 above)
Comment (josteink) [inline src/pal/tools/cppmunge/CMakeLists.txt]: If it's dead code anyway, I honestly don't think it's worth redoing this part of the PR. And until the PR to delete it is merged, this is still something stopping the build for us working on FreeBSD, so I'd like to keep it in.

Comment (jkotas) [inline src/pal/tools/cppmunge/CMakeLists.txt]: The PR to delete cppmunge is merged now.

**Comment by jkotas** [comment]
@josteink Could you please undo the cppmunge fix, and define _POSIX_C_SOURCE=200809L in CMakefile instead to fix the break? Thanks.

**Comment by josteink** [comment]
No problem. Will do.

**Comment by josteink** [comment]
Done. That certainly shortened the patch down :)

**Pull request merged**